### PR TITLE
InputText bug

### PR DESCRIPTION
**Pull request by ssaket**
1. Fix for https://github.com/BabylonJS/Babylon.js/issues/5471
2. Added mouse drag to support text highlighting

### CODE REVIEW

**Comment by ssaket** [comment]
Sure, Thanks for the update.

File: gui/src/2D/controls/inputText.ts [hunk 1]
```diff
@@ -800,9 +934,12 @@ export class InputText extends Control implements IFocusableControl {
         this._clickedCoordinate = coordinates.x;
         this._isTextHighlightOn = false;
         this._highlightedText = "";
+        this._cursorIndex = -1;
         if (this._host.focusedControl === this) {
             // Move cursor
             clearTimeout(this._blinkTimeout);
+            this._isPointerDown = true;
+            this._host._capturingControl[pointerId] = this;
```
[the remark this answers is not in the document]
Comment (ssaket) [reply gui/src/2D/controls/inputText.ts]: yes, that was my intention. Only when the input is focused, it should allow to drag. But, now seeing the other sources/inputText, I think it is allowed to click and drag the very first time. Will update 👍

File: gui/src/2D/controls/inputText.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export class InputText extends Control implements IFocusableControl {
     private _highlightedText = "";
     private _startHighlightIndex = 0;
     private _endHighlightIndex = 0;
+    private _cursorIndex = -1;
```
[the remark this answers is not in the document]
Comment (ssaket) [reply gui/src/2D/controls/inputText.ts]: when doing shift+arrow or mouse drag operation, _cursorIndex stores the starting index and _cursorOffset acts as current cursor pointer.

**Comment by deltakosh** [comment]
Do you mind fixing the conflicts?

**Comment by ssaket** [comment]
Hi @deltakosh, welcome back ! 
one question - where is _tempParentMeasure getting assigned ? I see only references in the code

**Comment by ssaket** [comment]
Doing few tests again. Please wait

**Comment by ssaket** [comment]
Done 👍

**Comment by deltakosh** [comment]
And merged! thanks a lot again @ssaket

**Pull request merged**